### PR TITLE
Using tensor methods in normalization.

### PR DESCRIPTION
**Pull request by adityaiitb**
Signed-off-by: Aditya Agrawal <aditya.iitb@gmail.com>

### CODE REVIEW

**Comment by dzier** [comment]
Is there a reason you made these individual file changes in separate commits and PRs. Any chance of merging to a single single PR so it is easier our resources for CI checking?

File: pyprof/prof/normalization.py [hunk 1]
```diff
@@ -28,49 +26,42 @@ def __init__(self, d):
...
+        self._op = op
 
         assert (op == "batch_norm")
         assert (len(args) >= 1)
         i = args[0]
         assert (i['type'] == "tensor")
 
-        self.shape = i['shape']
-        self.type = i['dtype']
+        self.inp = Tensor(i['shape'], i['dtype'])
```
Comment (dzier) [inline pyprof/prof/normalization.py:37]: `inp` is not a common name. Please elaborate for readability.

File: pyprof/prof/normalization.py [hunk 2]
```diff
@@ -28,49 +26,42 @@ def __init__(self, d):
...
-        return self.mod_
-
-    def elems(self):
-        return Utility.numElems(self.shape)
+        return self._mod
 
     def flops(self):
         # Variance algo-dependent, but this is a reasonable value.
-        return self.elems() * 8
+        return self.inp.size * 8
```
Comment (dzier) [inline pyprof/prof/normalization.py:55]: Why 8? Could this more controllable? Another named constant would help, i.e.

```
return self.inp.size * FLOP_MULTIPLY_FACTOR
```

In the future, do you think that this multiply factor may be fine tuned for the architecture or algorithm?

**Review by dzier — CHANGES_REQUESTED** [review]

File: pyprof/prof/normalization.py (hunk 2 above)
Comment (adityaiitb) [reply pyprof/prof/normalization.py:55]: The multiplication factor (8 in this case) depends on the operation (batch normalization in this case) and the direction in which it is called (forward or backward), so the constant is very specific to this op and would not be reused anywhere else (so hoisting it out would not be useful).

**Pull request closed without merging**